logic.js

```diff
@@ -0,0 +1,92 @@
+
+
+const Tweeter = function () {
+    
+    let _posts = [
+        {
+            text: "First post!",
+            id: "p1",
+            comments: [
+                { id: "c1", text: "First comment on first post!" },
+                { id: "c2", text: "Second comment on first post!!" },
+                { id: "c3", text: "Third comment on first post!!!" }
+            ]
+        },
+        {
+            text: "Aw man, I wanted to be first",
+            id: "p2",
+            comments: [
+                { id: "c4", text: "Don't wory second poster, you'll be first one day." },
+                { id: "c5", text: "Yeah, believe in yourself!" },
+                { id: "c6", text: "Haha second place what a joke." }
+            ]
+        }
+    ]
+    
+    let postIdCounter
+    let commentIdCounter
+    let getPosts = function () {
+        return _posts
+        
+    }
+    let idP = 2
+    let addPost = function (text) {
+        
+        idP = idP + 1
+        
+        let post = {
+            text: text,
+            id: 'p' + idP,
+            comments: []
+        }
+        _posts.push(post)
+    }
+    let removePost = function (postID) {
+        for (let i = 0; i < _posts.length; i++) {
+            if (postID === _posts[i].id){
+                _posts.splice((_posts.indexOf(_posts[i])), 1)
```

Comment:
_posts.indexOf(_posts[i]) is actually..... == i , so you don't need all this long line.... reading it is something like this: find me the index of the post in the i place